### PR TITLE
FCN example updates

### PR DESCRIPTION
**Pull request by lupesko**
## Description ##
Making the FCN example clear, easy to use and free of lint issues

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] Code is well-documented: example readme updated
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] All changes have test coverage - only up to the regular example testing

### Changes ###
- Fixed issue with Python3 that broke the example
- Fixed lint issues
- Added command line arg support for the image_segmentation file to make it easier to use
- Updated readme to make it more clear and easier to read
- Added reference to original paper

### CODE REVIEW

**Comment by lupesko** [comment]
@aaronmarkham @tornadomeet - please take a look, tx!

File: example/fcn-xs/README.md [hunk 1]
```diff
@@ -17,32 +18,36 @@ We have trained a simple fcn-xs model, the hyper-parameters are below:
...
-## How to train fcn-xs in mxnet
-#### Getting Started
+## Training the model
+
+### Step 1: setup pre-requisites
 
 - Install python package `Pillow` (required by `image_segment.py`).
 ```shell
-[sudo] pip install Pillow
+pip install Pillow
```
Comment (sandeep-krishnamurthy) [inline example/fcn-xs/README.md]: Will be good to give -U options for pip..

File: example/fcn-xs/image_segmentaion.py [hunk 2]
```diff
@@ -58,18 +88,35 @@ def get_data(img_path):
     return img
 
 def main():
-    fcnxs, fcnxs_args, fcnxs_auxs = mx.model.load_checkpoint(model_previx, epoch)
-    fcnxs_args["data"] = mx.nd.array(get_data(img), ctx)
+    """Module main execution"""
+    # Initialization variables - update to change your model and execution context
+    model_prefix = "FCN8s_VGG16"
+    epoch = 19
+    ctx = mx.cpu(0)
```
Comment (sandeep-krishnamurthy) [inline example/fcn-xs/image_segmentaion.py]: Will be useful to have a commented line here for #ctx = mx.gpu()
Uncomment this line to use GPU

File: example/fcn-xs/README.md (hunk 1 above)
Comment (lupesko) [reply example/fcn-xs/README.md]: Good point, updated to --upgrade for clarity

File: example/fcn-xs/image_segmentaion.py (hunk 2 above)
Comment (lupesko) [reply example/fcn-xs/image_segmentaion.py]: Updated.

**Comment by lupesko** [comment]
@sandeep-krishnamurthy: your comments addressed. Please review.

**Pull request merged**